frontend/src/stores/chatStore.ts

```diff
@@ -1,37 +1,124 @@
-import type { Chat } from '@/types/Chat';
+import type { Chat_List_Item } from '@/types/Chat';
+import type { SocketMessage } from '@/types/Sockets';
 import { getRequest } from '@/utils/apiRequests';
 import { defineStore } from 'pinia';
-import { useUserStore } from './userStore';
 
-export const useChatStore = defineStore("chats", {
+export const useChatStore = defineStore('chats', {
 	//  actions == data definitions
 	state: () => ({
-		allChats: <Chat[]>[],
+		dms: [] as Chat_List_Item[],
+		channels: [] as Chat_List_Item[],
+		subscribed: false,
+		initialized: false,
 	}),
-  // getters == computed values
+	// getters == computed values
 	getters: {
-		getMyChats: () => useUserStore().getMe.chats,
-        getAllChats: (state) => state.allChats,
+		getAllChats: (state) =>
+			[...state.dms, ...state.channels].sort((a, b) => a.id - b.id),
```

Comment:
the above can be typed liked this as well right? 
`[...state.dms, ...state.channels].sort((dm, channel) => dm.id - channel.id)`
so sort param 1 = a dm, param 2 = a channel?
then sorting on `dm.id - channel.d`
Could you maybe elaborate on what's happening here?

Reply:
there are 2 arrays in the component. DM's and Channels. 
the first part just puts them both in 1 array:
```ts
[...state.dms, ...state.channels]
```
the order in the new array at that point is the dms first, then the channels, which is not necessarily the order of their ID's (dms can be created later than channels etc)
so it's sorted on ID, because afterwards finding something is faster if the data is sorted.

the sorting happens on the joined array of dms and channels, where they have kinda lost their meaning of if they are a dm/channel, since they are now  both just in a list of "chats" (their type becomes irrelevant)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

`.sort()` is an Array method where (a,b) are the usual variable names. Just because it compares a to b. Javascript abstracts the sorting algorithm behind this, so you only supply a function to determine if `a` should be placed after `b`, or if `a` should be placed before `b` (the docs page is quite good... better than what I'm trying here..)

maybe I'm over explaining... I'll stop.. ask me if it's unclear :P

Reply:
hehe knowing that `[...state.dms, ...state.channels]` merges two array, it's clear thanks! :D